include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -371,113 +346,163 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _Up __u, _Cp __combine, _
 //------------------------------------------------------------------------
 // parallel_transform_scan - async pattern
 //------------------------------------------------------------------------
-template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _BinaryOperation, typename _InitType,
-          typename _LocalScan, typename _GroupScan, typename _GlobalScan>
-oneapi::dpl::__internal::__enable_if_device_execution_policy<
-    _ExecutionPolicy, oneapi::dpl::__par_backend_hetero::__future<typename _InitType::__value_type>>
-__parallel_transform_scan(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _BinaryOperation __binary_op,
-                          _InitType __init, _LocalScan __local_scan, _GroupScan __group_scan, _GlobalScan __global_scan)
+template <typename _GlobalScan, typename _Range2, typename _Range1, typename _Accessor, typename _Size>
+struct __global_scan_caller
 {
-    using _Policy = typename ::std::decay<_ExecutionPolicy>::type;
-    using _CustomName = typename _Policy::kernel_name;
-    using _Type = typename _InitType::__value_type;
-
-    using _LocalScanKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::_KernelName_t<__parallel_scan_local_kernel, _CustomName, _Range1,
-                                                                     _Range2, _BinaryOperation, _Type, _LocalScan,
-                                                                     _GroupScan, _GlobalScan>;
-    using _GlobalScanKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::_KernelName_t<__parallel_scan_global_kernel, _CustomName,
-                                                                     _Range1, _Range2, _BinaryOperation, _Type,
-                                                                     _LocalScan, _GroupScan, _GlobalScan>;
-    using _PropagateKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::_KernelName_t<__parallel_scan_propagate_kernel, _CustomName,
-                                                                     _Range1, _Range2, _BinaryOperation, _Type,
-                                                                     _LocalScan, _GroupScan, _GlobalScan>;
-
-    auto __n = __rng1.size();
-    assert(__n > 0);
+    __global_scan_caller(const _GlobalScan& __global_scan,
+                         const _Range2& __rng2,
```

Comment:
If I'm  not mistaken for a global scan functor we have the first range as an output range. So, `conts&` looks suspicious...
F.e. see
`operator()(_Item __item, _OutAcc& __out_acc, const _InAcc& __in_acc, const _WgSumsAcc& __wg_sums_acc, _Size __n,
`
https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h#L386

Reply:
This is usual to take the parameter by `const lvalue&` if we want to copy it into class